plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueClientUtil.java

```diff
@@ -13,57 +13,63 @@
-            RequestMetricCollector metricsCollector)
+            AwsCredentialsProvider credentialsProvider,
+            Optional<ExecutionInterceptor> requestHandler,
+            MetricPublisher metricPublisher)
     {
-        ClientConfiguration clientConfig = new ClientConfiguration()
-                .withMaxConnections(config.getMaxGlueConnections())
-                .withMaxErrorRetry(config.getMaxGlueErrorRetries());
-        AWSGlueAsyncClientBuilder asyncGlueClientBuilder = AWSGlueAsyncClientBuilder.standard()
-                .withMetricsCollector(metricsCollector)
-                .withClientConfiguration(clientConfig);
+        NettyNioAsyncHttpClient.Builder nettyBuilder = NettyNioAsyncHttpClient.builder()
+                .maxConcurrency(config.getMaxGlueConnections());
+        RetryPolicy.Builder retryPolicy = RetryPolicy.builder().numRetries(config.getMaxGlueErrorRetries());
+        ClientOverrideConfiguration.Builder clientOverrideConfiguration = ClientOverrideConfiguration.builder()
+                .addMetricPublisher(metricPublisher)
+                .retryPolicy(retryPolicy.build());
 
-        ImmutableList.Builder<RequestHandler2> requestHandlers = ImmutableList.builder();
+        ImmutableList.Builder<ExecutionInterceptor> requestHandlers = ImmutableList.builder();
         requestHandler.ifPresent(requestHandlers::add);
         config.getCatalogId().ifPresent(catalogId -> requestHandlers.add(new GlueCatalogIdRequestHandler(catalogId)));
-        asyncGlueClientBuilder.setRequestHandlers(requestHandlers.build().toArray(RequestHandler2[]::new));
+        clientOverrideConfiguration.executionInterceptors(requestHandlers.build());
+        GlueAsyncClientBuilder glueAsyncClientBuilder = GlueAsyncClient.builder()
+                .httpClient(nettyBuilder.build())
+                .overrideConfiguration(clientOverrideConfiguration.build());
 
         if (config.getGlueEndpointUrl().isPresent()) {
             checkArgument(config.getGlueRegion().isPresent(), "Glue region must be set when Glue endpoint URL is set");
-            asyncGlueClientBuilder.setEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(
-                    config.getGlueEndpointUrl().get(),
-                    config.getGlueRegion().get()));
+            glueAsyncClientBuilder
+                    .endpointOverride(URI.create(config.getGlueEndpointUrl().get()))
+                    .region(Region.of(config.getGlueRegion().get()));
         }
         else if (config.getGlueRegion().isPresent()) {
-            asyncGlueClientBuilder.setRegion(config.getGlueRegion().get());
+            glueAsyncClientBuilder.region(Region.of(config.getGlueRegion().get()));
         }
         else if (config.getPinGlueClientToCurrentRegion()) {
-            asyncGlueClientBuilder.setRegion(getCurrentRegionFromEC2Metadata().getName());
+            glueAsyncClientBuilder.region(Region.of(EC2MetadataUtils.getEC2InstanceRegion()));
```

Comment:
Is this call actually needed? According to the SDK documentation for `AwsClientBuilder.region()`:

> If this is not specified, the SDK will attempt to identify the defaults mode automatically using the following logic: [...] If running in EC2, check the EC2 metadata service for the region.

So it seems that we could deprecate the `hive.metastore.glue.pin-client-to-current-region` config option (remove it and mark with `@DefunctConfig`).

It also seems that we shouldn't use `EC2MetadataUtils` since it is in the `software.amazon.awssdk.regions.internal.util` package and the documentation says not to use it:

> Note: this is an internal API subject to change. Users of the SDK should not depend on this.

Reply:
yes, that is correct, should not be needed. Removed it.